src/gar.lua

```diff
@@ -1111,6 +1116,19 @@ function gar.allowDelegates(delegates, gatewayAddress)
 	}
 end
 
+function gar.isEligibleForArNSDiscount(from)
+	local gateway = gar.getGatewayUnsafe(from)
+	if gateway == nil or gateway.weights == nil then
```

Comment:
i would check that gateway is active as well. leaving gateways should not get discounts.

```suggestion
	if gateway == nil or gateway.weights == nil or gar.isGatewayLeaving(gateway) then
```

Reply:
ee8d8f3